compiler_opt/rl/unroll/config.py

```diff
@@ -0,0 +1,72 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Loop unroll training config."""
+
+import gin
+import tensorflow as tf
+from tf_agents.specs import tensor_spec
+from tf_agents.trajectories import time_step
+from compiler_opt.rl import feature_ops
+
+
+# pylint: disable=g-complex-comprehension
+@gin.configurable()
+def get_loop_unroll_signature_spec():
+  """Returns (time_step_spec, action_spec) for LLVM loop unroll."""
+  # LINT.IfChange
+  observation_spec = dict(
+      (key, tf.TensorSpec(dtype=tf.int64, shape=(), name=key)) for key in (
+          # Base feature
+          'loop_size',
+          'trip_count',
+          'is_innermost_loop',
+          'preheader_blocksize',
+          'bb_count',
+          'num_of_loop_latch',
+          'load_inst_count',
+          'store_inst_count',
+          'logical_inst_count',
+          'cast_inst_count'))
+  reward_spec = tf.TensorSpec(dtype=tf.float32, shape=(), name='reward')
+  time_step_spec = time_step.time_step_spec(observation_spec, reward_spec)
+  action_spec = tensor_spec.BoundedTensorSpec(
+      dtype=tf.int64, shape=(), name='unroll_count')
+
+  return time_step_spec, action_spec
+
+
+@gin.configurable
+def get_observation_processing_layer_creator(quantile_file_dir=None,
+                                             with_sqrt=True,
+                                             with_z_score_normalization=True,
+                                             eps=1e-8):
+  """Wrapper for observation_processing_layer."""
+  quantile_map = feature_ops.build_quantile_map(quantile_file_dir)
+
+  def observation_processing_layer(obs_spec):
```

Comment:
This layer expects how we want the "base feature" be fed into the network, where we can deal with different features (see [regalloc/config.py](https://github.com/google/ml-compiler-opt/blob/main/compiler_opt/rl/regalloc/config.py#L80) for example)
The `inlining_default` is logged as the input feature. The logging back the there is for monitoring issue.